### PR TITLE
[FIX] hr_holidays: modify validated leave

### PR DESCRIPTION
**Pull request by nim-odoo**
- Create and validate a leave
- Go to Time Off > Everyone
- In the Gantt view, move the leave

Nothing prevents the modification of the leave while is it validated.

opw-2179839

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -496,6 +496,12 @@ def _check_holidays(self):
                 raise ValidationError(_('The number of remaining time off is not sufficient for this time off type.\n'
                                         'Please also check the time off waiting for validation.'))
 
+    @api.constrains('date_from', 'date_from', 'employee_id')
```
Comment (MiquelRForgeFlow) [inline addons/hr_holidays/models/hr_leave.py]:
```suggestion
    @api.constrains('date_from', 'date_to', 'employee_id')
```
🤔

**Comment by nim-odoo** [comment]
@tivisse I had to modify some tests for this constraint. I don't really see the use case behind the creation of a leave directly in 'refuse' or 'validate' state. Indeed, the leave cannot be modified afterwards.

If it is really necessary, I could use a override in `write` instead, but the constrain seemed cleaner.

**Comment by tivisse** [comment]
Hi @nim-odoo 

Indeed you're right. Only auto-validated leaves are created directly in the "confirm" state, and then validated after the leave creation.

So once you've updated the .po file to manage the new translation, it's ok for me to merge this ;)

Thanks.

**Comment by nim-odoo** [comment]
robodoo r+

**Pull request closed without merging**